src/main/java/org/json/JSONObject.java

```diff
@@ -2886,4 +2889,33 @@ private static JSONException recursivelyDefinedObjectException(String key) {
             "JavaBean object contains recursively defined member variable of key " + quote(key)
         );
     }
+
+    /**
+     * For a prospective number, remove the leading zeros
+     * @param value prospective number
+     * @return number without leading zeros
+     */
+    private static String removeLeadingZerosOfNumber(String value){
+        char[] chars = value.toCharArray();
+        int leftMostUnsignedIndex = 0;
```

Comment:
`leftMostUnsignedIndex` could just be a boolean value. I'm not sure using an `int` here are repurposing it in the `for` initializer adds enough performance benefit over the readability of just using a boolean.

Reply:
Done. Code is more readable now.